rust/main/chains/hyperlane-sealevel/src/provider.rs

```diff
@@ -2,37 +2,110 @@ use std::sync::Arc;
 
 use async_trait::async_trait;
 use solana_sdk::signature::Signature;
-use solana_transaction_status::EncodedTransaction;
+use solana_transaction_status::{
+    option_serializer::OptionSerializer, EncodedTransaction, EncodedTransactionWithStatusMeta,
+    UiMessage, UiTransaction, UiTransactionStatusMeta,
+};
 
 use hyperlane_core::{
     BlockInfo, ChainCommunicationError, ChainInfo, ChainResult, HyperlaneChain, HyperlaneDomain,
-    HyperlaneProvider, HyperlaneProviderError, TxnInfo, TxnReceiptInfo, H256, H512, U256,
+    HyperlaneProvider, HyperlaneProviderError, NativeToken, TxnInfo, TxnReceiptInfo, H256, H512,
+    U256,
 };
 
 use crate::error::HyperlaneSealevelError;
 use crate::utils::{decode_h256, decode_h512, decode_pubkey};
 use crate::{ConnectionConf, SealevelRpcClient};
 
+/// Exponent value for atto units (10^-18).
+const ATTO_EXPONENT: u32 = 18;
```

Comment:
noting that this is also defined in `hyperlane-cosmos` in `provider.rs`

Reply:
Refactored.